pa_bin/performance-analyzer-agent

```diff
@@ -15,14 +15,23 @@ else
 fi
 
 if [ -z "$2" ]; then
-  if [ -z "$JAVA_HOME" ]; then
-    echo "JAVA_HOME needs to be set or passed in as the second parameter."
-    exit 1
+  # Check if OPENSEARCH_JAVA_HOME is present
+  if [ -z "$OPENSEARCH_JAVA_HOME" ]; then
+      # If OPENSEARCH_JAVA_HOME is not present, check JAVA_HOME next
+      if [ -z "$JAVA_HOME" ]; then
+        echo "OPENSEARCH_JAVA_HOME / JAVA_HOME needs to be set or passed in as the second parameter."
+        exit 1
+      fi
+  else
+    JAVA_HOME=$OPENSEARCH_JAVA_HOME
   fi
 else
   JAVA_HOME=$2
 fi
 
+echo "Using JAVA_HOME: $JAVA_HOME"
+export JAVA_HOME=$JAVA_HOME
```

Comment:
Interestingly, the `JAVA_HOME` is ignored right now and it works only because `bin/performance-analyzer-rca` picks `java` command line from the `$PATH`. It is easy to check and reproduce running fe:
```
JAVA_HOME=/this/path/does/not/exists OPENSEARCH_HOME=. ./pa_bin/performance-analyzer-agent
```

or 

```
./pa_bin/performance-analyzer-agent . /this/path/does/not/exists
```

Adding `export JAVA_HOME=$JAVA_HOME` forces  `bin/performance-analyzer-rca` to use the right JVM location from the environment: either `OPENSEARCH_JAVA_HOME` or `JAVA_HOME`

Reply:
🤦